src/Relation/Binary/Rewriting.agda

```diff
@@ -0,0 +1,135 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Concepts from rewriting theory
+-- Definitions are based on "Term Rewriting Systems" by J.W. Klop
+------------------------------------------------------------------------
+
+{-# OPTIONS --with-K --safe #-}
+
+module Relation.Binary.Rewriting where
+
+open import Agda.Builtin.Equality using (_≡_ ; refl)
+open import Data.Product using (_×_ ; ∃ ; _,_ ; proj₂)
+open import Data.Empty
+open import Data.Sum as Sum using (_⊎_)
+open import Level
+open import Relation.Binary.Core
+open import Relation.Binary.Construct.Closure.Transitive
+open import Relation.Binary.Construct.Closure.ReflexiveTransitive
+open import Relation.Binary.Construct.Closure.Symmetric
+open import Relation.Nullary
+
+-- The following definitions are taken from Klop [5]
+IsNormalForm : ∀ {A : Set} → {ℓ : Level} → {r : Rel A ℓ} → (a : A) → Set _
```

Comment:
Is there a reason why A is not universe polymorphic? If made universe polymorphic then following standard library conventions this this type would be written as `∀ {a ℓ} {A : Set a} →Rel A ℓ → A → Set _`.

(These apply to many of the definitions below as well)

Reply:
Note that I have made the `Rel` argument explicit as it can't be inferred from the `A` parameter. Indeed you seem to need to provide it explicitly everywhere below.